app/Mage.php

```diff
@@ -527,15 +527,15 @@ public static function getModel($modelClass = '', $arguments = array())
      *
      * @param   string $modelClass
      * @param   array $arguments
-     * @return  Mage_Core_Model_Abstract
+     * @return  Mage_Core_Model_Abstract|false
```

Comment:
PHPStan complains about this change but I did it because the method is using `getModel` and it is possible that this method returns false (in case the given `$modelClass` is unknown). In turn it is possible that "false" is registered and `getSingleton` returns false as well. What can be done about this?

Reply:
There's some "magic" taking place because `Mage::getModel()` also returns `Mage_Core_Model_Abstract|false` yet phpstan doesn't complain about its usage. I don't see it in the baseline file either.

Reply:
I know what's happening, the error is only triggered for `getSingleton` calls with a variable as `$modelClass`.

Reply:
I'm not sure how to proceed, I don't like the possibility of `getModel()` and in turn `getSingleton()`  to return false to begin with (in contrast to something like `helper()`) but changing that behavior shouldn't be in the scope of this PR and would constitute a change for 20. I'm fully aware that returning false is somewhat of an edge case. 
I could revert the added `|false`. Otherwise the objected lines by PHPStan would need an added type check I guess.

Reply:
[macopedia/phpstan-magento1](https://github.com/macopedia/phpstan-magento1) has some extension code for PHPStan that tells it whether `getModel` or `getSingleton` will return `false` depending if the model actually exists. These errors are only from cases where a dynamic model class is given (PHPStan can't resolve that so it assumes false is a possible value), I'm not sure if adding a `@var` doc line above those calls would solve this..

Reply:
I tried to fix the errors with annotations but it solved only half the errors. I therefore have decided to revert my change to the return annotation of `getSingleton()`. While I still think that the return type must be different, this and subsequent changes (possibly to the PHPStan plugin as well) is not part of scope of this PR.